neworigins/map.cpp

```diff
@@ -0,0 +1,1378 @@
+// START A3HEADER
```

Comment:
If this is identical to one of the map.cpps in other directories, maybe we need to rethink how these are split out to avoid duplicating code?  And similarly for some of the other files in here.

Reply:
i am not very familiar with codebase, it has exactly same structure as Havilah